src/fsharp/TypeChecker.fs

```diff
@@ -8644,13 +8646,16 @@ and TcItemThen cenv overallTy env tpenv (item, mItem, rest, afterResolution) del
                                 assert (isNull(box fittedArgs.[currentIndex]))
                                 fittedArgs.[currentIndex] <- List.item currentIndex args // grab original argument, not item from the list of named parameters
                                 currentIndex <- currentIndex + 1
-                            else
-                                let caseName = 
-                                    match item with
-                                    | Item.UnionCase(uci, _) -> uci.Name
-                                    | Item.ExnCase tcref -> tcref.DisplayName
-                                    | _ -> failwith "impossible"
-                                error(Error(FSComp.SR.tcUnionCaseConstructorDoesNotHaveFieldWithGivenName(caseName, id.idText), id.idRange))
+                            else                                
+                                match item with
+                                | Item.UnionCase(uci, _) ->
+                                    error(Error(FSComp.SR.tcUnionCaseConstructorDoesNotHaveFieldWithGivenName(uci.Name, id.idText), id.idRange))
+                                | Item.ExnCase tcref ->
+                                    error(Error(FSComp.SR.tcExceptionConstructorDoesNotHaveFieldWithGivenName(tcref.DisplayName, id.idText), id.idRange))
+                                | Item.ActivePatternResult(_,_,_,_) ->
+                                    error(Error(FSComp.SR.tcActivePatternsDoNotHaveFields(), id.idRange))
+                                | _ -> 
+                                    error(Error(FSComp.SR.tcConstructorDoesNotHaveFieldWithGivenName(id.idText), id.idRange))
```

Comment:
Should we still report that this error message might need to be refined (in case it happens)? Would this be an advantage of `failwith "impossible"` here?

Reply:
failwith "impossible" is arguably the worst you can do

Reply:
Yes ideally the code would be written in a way to not compile when written incorrectly, but if that cannot be achieved imho throwing is better than doing/reporting potentially incorrect messages

Reply:
Actually I think even having a confusing error message is better than what existed. The current state of the world (without @forki's changes) paints the whole editor red if you use the incorrect code:

<img width="439" alt="screen shot 2018-10-09 at 07 28 16" src="https://user-images.githubusercontent.com/6309070/46676235-ed33e680-cb94-11e8-8b49-2bee1c809bab.png">

These internal errors are critical to root out given this, since it's pretty much undiagnosable where the user error is.

Reply:
Ok but imho we still should make sure the error message indicates this is not implemented and something which needs to be fixed if it occurs.
Otherwise users get wrong messages on their code which is arguably worse than a compiler crash

Reply:
we don't give a wrong message. We just give a very generic one